File: android/src/com/dontflymoney/view/LoginActivity.java
package com.dontflymoney.view;

import org.json.JSONException;
import org.json.JSONObject;

import android.os.Bundle;
import android.view.View;

import com.dontflymoney.api.Request;
import com.dontflymoney.api.Step;


public class LoginActivity extends SmartActivity
{
	public LoginActivity()
	{
		init(this, R.layout.activity_login, R.menu.login);
	}
	
	
	
	
	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);
		
		if (Authentication.IsLoggedIn())
		{
			redirect(AccountsActivity.class);
		}
	}	

	
	
	public void login(View view)
	{
		Request request = new Request(this, "Users/Login");
		
		request.AddParameter("email", getValue(R.id.email));
		request.AddParameter("password", getValue(R.id.password));
		
		request.Post();
	}
	
	@Override
	protected void HandleSuccess(JSONObject result, Step step) throws JSONException
	{
		String ticket = result.getString("data");
		Authentication.Set(ticket);
		
		redirect(AccountsActivity.class);
	}
	

}

File: android/DFM/src/main/kotlin/com/darakeon/activities/LoginActivity.kt
package com.dontflymoney.view

import android.view.View
import com.dontflymoney.activityObjects.LoginStatic
import com.dontflymoney.api.InternalRequest
import com.dontflymoney.api.Step
import com.dontflymoney.baseactivity.SmartActivity
import org.json.JSONException
import org.json.JSONObject


class LoginActivity : SmartActivity<LoginStatic>(LoginStatic) {
    override fun contentView(): Int {
        return R.layout.login
    }

    override val isLoggedIn: Boolean
        get() = false


    fun login(view: View) {
        val request = InternalRequest(this, "Users/Login")

        request.AddParameter("email", form.getValue(R.id.email))
        request.AddParameter("password", form.getValue(R.id.password))

        request.Post()
    }

    @Throws(JSONException::class)
    override fun HandleSuccess(data: JSONObject, step: Step) {
        val ticket = data.getString("ticket")
        Authentication.Set(ticket)
        navigation.redirect(AccountsActivity::class.java)
    }


}
